core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamProcessingStrategy.java

```diff
@@ -107,43 +119,60 @@ public ReactiveProcessor onProcessor(ReactiveProcessor processor) {
     }
   }
 
-  private ReactiveProcessor proactor(ReactiveProcessor processor, Scheduler scheduler) {
-    // TODO MULE-17079 Remove this policyMode flag.
-    // This is to avoid the performance degradation introduced by MULE-17060 when using CPU_INTENSIVE processors in policies,
-    // until MULE-17079 is done.
-    if (policyMode) {
-      return publisher -> withRetry(scheduleProcessor(processor, scheduler, from(publisher))
-          .subscriberContext(ctx -> ctx.put(PROCESSOR_SCHEDULER_CONTEXT_KEY, scheduler)), scheduler);
+  protected ReactiveProcessor proactor(ReactiveProcessor processor, ScheduledExecutorService scheduler) {
+    LOGGER.debug("Doing proactor() for {} on {}. maxConcurrency={}, parallelism={}, subscribers={}", processor, scheduler,
+                 maxConcurrency, getParallelism(), subscribers);
+
+    final ScheduledExecutorService retryScheduler =
+        new RejectionCallbackExecutorServiceDecorator(scheduler, getCpuLightScheduler(),
+                                                      () -> onRejected(scheduler),
+                                                      () -> lastRetryTimestamp.set(MIN_VALUE),
+                                                      ofMillis(SCHEDULER_BUSY_RETRY_INTERVAL_MS));
+
+    // FlatMap is the way reactor has to do parallel processing. Since this proactor method is used for the processors that are
+    // not CPU_LITE, parallelism is wanted when the processor is blocked to do IO or doing long CPU work.
+    if (maxConcurrency <= getParallelism() * subscribers) {
+      // If no concurrency needed, execute directly on the same Flux
+      return publisher -> scheduleProcessor(processor, retryScheduler, from(publisher))
+          .subscriberContext(ctx -> ctx.put(PROCESSOR_SCHEDULER_CONTEXT_KEY, scheduler));
+    } else if (maxConcurrency == MAX_VALUE) {
+      if ((processor instanceof InterceptedReactiveProcessor)
+          && sdkOperationClass.isAssignableFrom(((InterceptedReactiveProcessor) processor).getProcessor().getClass())) {
+        // For no limit, the java SDK already does a flatMap internally, so no need to do an additional one here
```

Comment:
Seems like we will need to add a comment on that code regarding it's connection to this, so that no one changes it without knowing the consequences.